### PR TITLE
Feature/air 51 asdocs

### PR DESCRIPTION
**Pull request by ajwfrost**
Hi
Updated ActionScript APIs for AIR 51.0 are in this branch. Likely to issue the preview release tomorrow so is it possible to get this merged in and published please? We're referencing the "what's new" page in the release notes..
thanks
  Andrew

### CODE REVIEW

**Review by marchbold — APPROVED** [review]

File: static/reference/actionscript/3.0/Function.html [hunk 1]
```diff
@@ -222,6 +231,28 @@
...
+</tr>
+</table>
+<div class="detailBody">
+<code>declaration:<a href="String.html">String</a></code>&nbsp;&nbsp;[read-only]<p>
+
+     Returns a string representing the function declaration from the source code.
+
+     </p><p>This can be used to find out naming and parameter details, and also determine what kind of function
+
+     the object is (anonymous function vs member fnction, etc).</p>
```
Comment (Adolio) [inline static/reference/actionscript/3.0/Function.html:249]: nit: Small typo here `function`

**Review by itlancer — COMMENTED** [review]
May be wrong AIR 52 (not 51) mention

File: static/reference/actionscript/3.0/Function.html (hunk 1 above)
Comment (ajwfrost) [reply static/reference/actionscript/3.0/Function.html:249]: Good spot, thanks!

File: static/reference/actionscript/3.0/flash/display/InteractiveObject.html [hunk 2]
```diff
@@ -3779,6 +3787,61 @@
...
+<br>
+<p></p>
+<table border="0" cellspacing="0" cellpadding="0">
+<tr>
+<td valign="top" style="white-space:nowrap"><b>Language version:&nbsp;</b></td><td>ActionScript 3.0  </td>
+</tr>
+</table>
+<table border="0" cellspacing="0" cellpadding="0">
+<tr>
+<td valign="top" style="white-space:nowrap"><b>Runtime version:&nbsp;</b></td><td>AIR&nbsp;52
```
[the remark this answers is not in the document]
Comment (ajwfrost) [reply static/reference/actionscript/3.0/flash/display/InteractiveObject.html:3809]: Yes not sure why we put 52!

File: static/reference/actionscript/3.0/Array.html [hunk 3]
```diff
@@ -1259,6 +1279,61 @@
...
+</td>
+</tr>
+</table>
+<p></p><p>
+
+     Checks whether the array includes the item that is passed in.
+
+     Each element is checked in turn to see if it matches the item that is passed in,
+
+     using the standard comparison operator (i.e. similar to <code>if (array[0] == item)</code>).
```
[the remark this answers is not in the document]
Comment (ajwfrost) [reply static/reference/actionscript/3.0/Array.html:1310]: Good idea

File: static/reference/actionscript/3.0/Vector.html [hunk 4]
```diff
@@ -1044,6 +1062,49 @@
...
+<table border="0" cellspacing="0" cellpadding="0">
+<tr>
+<td valign="top" style="white-space:nowrap"><b>Runtime version:&nbsp;</b></td><td>AIR&nbsp;51.0
+</td>
+</tr>
+</table>
+<p></p><p>
+     Checks whether the vector includes the item that is passed in.
+     Each element is checked in turn to see if it matches the item that is passed in,
+     using the standard comparison operator (i.e. similar to <code>if (vector[0] == item)</code>).
```
[the remark this answers is not in the document]
Comment (ajwfrost) [reply static/reference/actionscript/3.0/Vector.html:1088]: Good idea

File: static/reference/actionscript/3.0/flash/events/MouseEvent.html [hunk 5]
```diff
@@ -2362,6 +2367,60 @@
...
+<code>public static const MOUSE_WHEEL_HORIZONTAL:<a href="../../String.html">String</a> = "mouseWheelHorizontal"</code>
+<p></p>
+<table border="0" cellspacing="0" cellpadding="0">
+<tr>
+<td valign="top" style="white-space:nowrap"><b>Language version:&nbsp;</b></td><td>ActionScript 3.0    </td>
+</tr>
+</table>
+<table border="0" cellspacing="0" cellpadding="0">
+<tr>
+<td valign="top" style="white-space:nowrap"><b>Runtime version:&nbsp;</b></td><td>AIR&nbsp;52
```
[the remark this answers is not in the document]
Comment (ajwfrost) [reply static/reference/actionscript/3.0/flash/events/MouseEvent.html:2386]: Again.. oops, correcting that to 51!

**Comment by ajwfrost** [comment]
I've just made the updates to the AS3 source code for the above feedback, will see whether we have any other errors in the documentation and then re-generate the HTML and push a further update in a few days...

thanks

File: static/reference/actionscript/3.0/Array.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (Adolio) [reply static/reference/actionscript/3.0/Array.html:1310]: Or simply `array.indexOf(item) != -1`

**Comment by itlancer** [comment]
@ajwfrost 
Few more comments:
1) Couldn't find `NetworkInfo::disableNetworkChanges` documentation.
2) Perhaps it should be documented at the same time:
https://github.com/airsdk/Adobe-Runtime-Support/issues/2809
https://github.com/airsdk/Adobe-Runtime-Support/issues/2492#issuecomment-1729854016

**Comment by itlancer** [comment]
@ajwfrost 
And may be need to add `Font::registerFontFromByteArray()` to "What's New" and PDF Release Notes. And may be here https://github.com/airsdk/Adobe-Runtime-Support/discussions/3081

**Comment by ajwfrost** [comment]
Hi

We've got some further updates yes, just need to pick out the files that need to be updated and then create the next pull request.

That Font update is deliberately missing .. we added the AS3 API in preparation for the feature but I'm actually not sure yet whether it's feasible, it doesn't look like we have the source code from Adobe for font embedding within the compiler - it's pulled in from a separate library that I think was created by a different team and may no longer be available to us. So, not sure whether those "dynamic loading of font files" features will be viable, hence keeping them 'private' for now.

thanks

**Comment by ajwfrost** [comment]
> https://airsdk.dev/reference/actionscript/3.0/flash/net/NetworkInfo.html#findInterfaces()
NetworkInfo::findInterfaces() has corrupted link to NetworkInterface. Just try to click on NetworkInterface link.

Looks like a more generic issue with handling of 'Vector' types... e.g. Stage.stage3Ds has the same problem.

**Pull request merged**